PYIC-1834: Contra-indicator storage core stub

Wynndow opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Contra-indicator storage core stub

### Why did it change

Adds a lambda that will be deployed to dev and build accounts that will
allow interaction with the contra-indicator storage system for dev and
testing purposes.

It provides a web page that can be used to submit CIs and a user ID to
the system, wrapped in a signed JWT. It will also retrieve the CIs for a
given user ID.

This makes an assumption of the interface for the getContraindicator
lambda, which has not been built yet, and so may need adjusting in
future.

### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-1834](https://govukverify.atlassian.net/browse/PYIC-1834)

### Screenshots
![Screenshot 2022-08-22 at 13 30 27](https://user-images.githubusercontent.com/13836290/185921577-6225abf2-3cd7-42a4-95b4-4604cdbc323e.png)


![Screenshot 2022-08-22 at 13 30 42](https://user-images.githubusercontent.com/13836290/185921605-8d7b0ca9-366e-4442-8e61-686465b8cdd1.png)

Wynndow committed:
PYIC-1834: Contra-indicator storage core stub

Adds a lambda that will be deployed to dev and build accounts that will
allow interaction with the contra-indicator storage system for dev and
testing purposes.

It provides a web page that can be used to submit CIs and a user ID to
the system, wrapped in a signed JWT. It will also retrieve the CIs for a
given user ID.

This makes an assumption of the interface for the getContraindicator
lambda, which has not been built yet, and so may need adjusting in
future.

dannyhayes-gds commented on lambdas/contra-indicator-storage-core-stub/src/main/java/uk/gov/di/ipv/core/contraindicatorstoragecorestub/domain/GetCiRequest.java:
```diff
@@ -0,0 +1,17 @@
+
+@ExcludeFromGeneratedCoverageReport
+@Getter
+@SuppressWarnings("java:S116") // Field names should comply with a naming convention
```
We previously used the Jackson @JsonProperty annotation here to handle the snake-case var names. I guess the downside is introducing another gradle dependency but would avoid having to suppress the warning which might be nicer?


But not too fussed about this so happy for you to decide what seems better.

Wynndow replied:
Yeah, I was trying to not use Jackson. The auth team found a lot of speed up in their lambdas by switching from Jackson to Gson.
I think I've just found something that can do a similar thing to `@JsonProperty` though so I might try adding it in.

dannyhayes-gds replied:
Ah okay yeah thats fair enough, happy to leave out jackson then 👍

Wynndow committed:
PYIC-1834: Use `@SerializedName`

We can use this annotation to ensure that the Java object field names
are consistent with the usual patterns.

Wynndow committed:
PYIC-1834: Make evidence an array

The evidence property of the VC should be an array
